File: python-backend/src/app/internal_dto/run_status_update_request.py
from typing import Any

from pydantic import BaseModel, Field


class RunStatusUpdateRequest(BaseModel):
    """Python 回写 Java 的 run 状态请求。"""

    status: str = Field(..., description="要回写的 run 状态")
    current_stage: str | None = Field(default=None, alias="currentStage", description="当前执行阶段")
    result_summary: str | None = Field(default=None, alias="resultSummary", description="运行结果摘要")
    clarification_payload: dict[str, Any] | None = Field(default=None, alias="clarificationPayload", description="需要用户补充的信息")
    approval_payload: dict[str, Any] | None = Field(default=None, alias="approvalPayload", description="待用户确认的审批包")
    error_code: str | None = Field(default=None, alias="errorCode", description="失败错误码")
    error_message: str | None = Field(default=None, alias="errorMessage", description="失败错误信息")

File: java-backend/common/src/main/java/com/airesumeforge/common/dto/agent/internal/request/RunStatusUpdateRequest.java
package com.airesumeforge.common.dto.agent.internal.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

/**
 * Python 回写 Agent run 状态请求
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RunStatusUpdateRequest {

    /**
     * run 状态：RUNNING / WAITING_USER / WAITING_CONFIRM / FAILED 等
     */
    @NotBlank(message = "status 不能为空")
    private String status;

    /**
     * 当前阶段编码
     */
    private String currentStage;

    /**
     * 当前 run 的结果摘要
     */
    private String resultSummary;

    /**
     * Clarifier 追问 payload
     */
    private Map<String, Object> clarificationPayload;

    /**
     * ApprovalPackager 审批 payload
     */
    private Map<String, Object> approvalPayload;

    /**
     * 失败时的错误信息
     */
    private String errorMessage;
}